backend/src/main/kotlin/org/loculus/backend/controller/FilesController.kt

```diff
@@ -0,0 +1,92 @@
+package org.loculus.backend.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.apache.http.HttpStatus
+import org.loculus.backend.api.AccessionVersion
+import org.loculus.backend.api.FileIdAndUrl
+import org.loculus.backend.auth.AuthenticatedUser
+import org.loculus.backend.auth.HiddenParam
+import org.loculus.backend.auth.User
+import org.loculus.backend.service.files.FilesDatabaseService
+import org.loculus.backend.service.files.S3Service
+import org.loculus.backend.service.groupmanagement.GroupManagementPreconditionValidator
+import org.loculus.backend.service.submission.AccessionPreconditionValidator
+import org.loculus.backend.service.submission.SubmissionDatabaseService
+import org.loculus.backend.utils.Accession
+import org.springframework.http.ResponseEntity
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+
+@RestController
+@RequestMapping("/files")
+@Validated
+@SecurityRequirement(name = "bearerAuth")
+class FilesController(
+    private val filesDatabaseService: FilesDatabaseService,
+    private val s3Service: S3Service,
+    private val groupManagementPreconditionValidator: GroupManagementPreconditionValidator,
+    private val submissionDatabaseService: SubmissionDatabaseService,
+    private val accessionPreconditionValidator: AccessionPreconditionValidator,
+) {
+
+    @GetMapping("/get/{accession}/{version}/{fileCategory}/{fileName}")
+    fun getFileDownloadUrl(
+        @HiddenParam user: User,
+        @PathVariable accession: Accession,
+        @PathVariable version: Long,
+        @PathVariable fileCategory: String,
+        @PathVariable fileName: String,
+    ): ResponseEntity<Void> {
+        val accessionVersion = AccessionVersion(accession, version)
+        val fileId = submissionDatabaseService.getFileId(accessionVersion, fileCategory, fileName)
```

Comment:
This seems to require `fileName`s to be unique for a given `accessionVersion` and `fileCategory` - is this checked for by the backend at point of processing the submission?

Should one be able to just get a file by `fileId` as well? Not the complicated `{accession}/{version}/{fileCategory}/{fileName}`?

Reply:
> Should one be able to just get a file by fileId as well? 

I don't see a need for this. At the moment file-ids are just a Loculus convenience feature - they are not something we plan to surface to users. That seems good to me. If you want a unique string to represent your file you have `{accession_version}/{fileCategory}/{fileName}`